dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-fee-record-payment-entity-groups-to-fee-record-payment-groups.ts

```diff
@@ -1,56 +1,123 @@
-import { FeeRecordEntity, FeeRecordStatus } from '@ukef/dtfs2-common';
+import { orderBy } from 'lodash';
+import { FEE_RECORD_STATUS, FeeRecordEntity, FeeRecordStatus } from '@ukef/dtfs2-common';
 import { FeeRecordPaymentEntityGroup, calculateTotalCurrencyAndAmount } from '../../../../../helpers';
 import { mapPaymentEntityToPayment } from '../../../../../mapping/payment-mapper';
 import { mapFeeRecordEntityToFeeRecord } from '../../../../../mapping/fee-record-mapper';
-import { FeeRecordPaymentGroup } from '../../../../../types/utilisation-reports';
+import { FeeRecordPaymentGroup, FeeRecordReconciledByUser } from '../../../../../types/utilisation-reports';
+import { TfmUsersRepo } from '../../../../../repositories/tfm-users-repo';
 
+/**
+ * Gets the status for a fee record payment entity group
+ * @param feeRecordEntitiesInGroup - The fee record entities in a group
+ * @returns The status of the group
+ */
 const getStatusForGroupOfFeeRecords = (feeRecordEntitiesInGroup: FeeRecordEntity[]): FeeRecordStatus => {
   if (feeRecordEntitiesInGroup.some((feeRecordEntity) => feeRecordEntity.status === 'READY_TO_KEY')) {
     return 'READY_TO_KEY';
   }
-
   return feeRecordEntitiesInGroup[0].status;
 };
 
+/**
+ * Gets the most recently reconciled fee record from a list of fee records
+ * @param feeRecordsAtReconciledStatus - The fee records (with status RECONCILED)
+ * @returns The most recently reconciled fee record
+ */
+const getMostRecentlyReconciledFeeRecord = (feeRecordsAtReconciledStatus: FeeRecordEntity[]): FeeRecordEntity => {
```

Comment:
Can we add a unit test for this please?

Reply:
They are all standalone functions that should have unit tests, can we add them in please.

```ts
getStatusForGroupOfFeeRecords
getMostRecentlyReconciledFeeRecord
getReconciledByUserForFeeRecord
getReconciliationDataForFeeRecords
```

We should have test coverage so that if one of these functions changes, a test for that function in particular, will fail. Rather than having a test for one massive function.

It can also help simplify the default mapping function

Reply:
See [below](https://github.com/UK-Export-Finance/dtfs2/pull/3417#discussion_r1729020528)